assets/js/common_utils.js

```diff
@@ -410,3 +410,67 @@ export function convertToFloat16OrUint16Array(fp32_array) {
     }
     return fp16_array;
 }
+
+// Create a new ORT ML Tensor from the given parameters.
+export async function createMlTensor(mlContext, dataType, dims, writable, readable) {
+    const mlTensor = await mlContext.createTensor({ dataType, shape: dims, writable, readable });
+    // eslint-disable-next-line no-undef
+    return ort.Tensor.fromMLTensor(mlTensor, { dataType, dims });
+}
+
+// Normalize the buffer size so that it fits the 128-bits (16 bytes) alignment.
+const calcNormalizedBufferSize = size => Math.ceil(Number(size) / 16) * 16;
+
+// Create a new ORT GPU Tensor from the given parameters.
+export function createGpuTensor(device, dataType, dims, bufferSize) {
+    const gpuBuffer = device.createBuffer({
+        usage: GPUBufferUsage.COPY_SRC | GPUBufferUsage.COPY_DST | GPUBufferUsage.STORAGE,
+        size: calcNormalizedBufferSize(bufferSize),
+    });
+    // eslint-disable-next-line no-undef
+    return ort.Tensor.fromGpuBuffer(gpuBuffer, { dataType, dims });
+}
+
+// Download an ML tensor into a pre-allocated target buffer.
+export async function downloadMlTensor(mlContext, mlTensor, targetBuffer) {
```

Comment:
The naming is the context of a web demo is a little confusing because if someone saw the verb "download", their first expectation is that it would download it from the web. How about `readBackMLTensor` or ~~`downloadMLTensorFromGPU`~~ *(not specific to GPU since NPU's also exist)* or `downloadMLTensorFromDevice`?

Reply:
OK, I vote `readBackMLTensor`. :)